utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -60,31 +56,74 @@ pub struct TryRuntimeCmd {
 		default_value = "Interpreted"
 	)]
 	pub wasm_method: WasmExecutionMethod,
+
+	/// The state to use to run the migration.
+	#[structopt(subcommand)]
+	pub state: State,
 }
 
 /// The state to use for a migration dry-run.
-#[derive(Debug)]
+#[derive(Debug, structopt::StructOpt)]
 pub enum State {
-	/// A snapshot. Inner value is a file path.
-	Snap(String),
+	/// Use a snapshot as state to run the migration.
+	Snap {
+		#[structopt(flatten)]
+		cache_params: CacheParams,
+	},
+
+	/// Use a live chain to run the migration.
+	Live {
+		/// An optional cache file to WRITE to. Not cached if set to `None`.
+		#[structopt(short, long)]
+		cache_file: Option<CacheParams>,
+
+		/// The block number at which to connect. Can be either a number or a hash (staring with `0x`) Will be latest finalized head if not provided.
+		#[structopt(short, long, multiple = false)]
+		block_at: Option<BlockNumberOrHash>,
+
+		/// The modules to scrape. If empty, entire chain state will be scraped.
+		#[structopt(short, long, require_delimiter = true)]
+		modules: Option<Vec<String>>,
+
+		/// The url to connect to.
+		#[structopt(default_value = "http://localhost:9933", parse(try_from_str = parse_url))]
+		url: String,
+	},
+}
 
-	/// A live chain. Inner value is the HTTP uri.
-	Live(String),
+fn parse_url(s: &str) -> Result<String, &'static str> {
+	match s.get(..7) {
```

Comment:
I think `s.starts_with("http://")`is more readable in this case

Reply:
I changed the code to use starts_with.